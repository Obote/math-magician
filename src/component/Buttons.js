import React, { useState } from 'react';
import calculate from '../logic/calculate';

function Buttons() {
  const [calculatorData, setCalculatorData] = useState({
    total: null,
    next: null,
    operation: null,
  });

  const handleButtonClick = (buttonName) => {
    const newData = calculate(calculatorData, buttonName);
    setCalculatorData(newData);
  };

  const renderButtons = () => {
    const buttonValues = [
      'AC',
      '+/-',
      '%',
      'รท',
      '9',
      '8',
      '7',
      'x',
      '6',
      '5',
      '4',
      '-',
      '3',
      '2',
      '1',
      '+',
      '0',
      '.',
      '=',
    ];

    return buttonValues.map((value) => (
      <input
        type="button"
        value={value}
        className={`button ${value === '0' ? 'double' : ''} ${
          ['+', '-', 'x', 'รท', '='].includes(value) ? 'orange' : ''
        }`}
        key={value}
        onClick={() => handleButtonClick(value)}
      />
    ));
  };

  return (
    <div className="calc">
      <input
        type="text"
        placeholder="0"
        id="answer"
        value={calculatorData.next || calculatorData.total || '0'}
        readOnly
      />
      {renderButtons()}
    </div>
  );
}

export default Buttons;
