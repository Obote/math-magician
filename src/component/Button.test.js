import React from 'react';
import renderer from 'react-test-renderer';
import Buttons from './Buttons';

test('should render', () => {
  const tree = renderer.create(<Buttons />).toJSON();
  expect(tree).toMatchSnapshot();
});
