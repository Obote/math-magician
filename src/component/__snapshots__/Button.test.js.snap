// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<div
  className="calc"
>
  <input
    id="answer"
    placeholder="0"
    readOnly={true}
    type="text"
    value="0"
  />
  <input
    className="button  "
    onClick={[Function]}
    type="button"
    value="AC"
  />
  <input
    className="button  "
    onClick={[Function]}
    type="button"
    value="+/-"
  />
  <input
    className="button  "
    onClick={[Function]}
    type="button"
    value="%"
  />
  <input
    className="button  orange"
    onClick={[Function]}
    type="button"
    value="รท"
  />
  <input
    className="button  "
    onClick={[Function]}
    type="button"
    value="9"
  />
  <input
    className="button  "
    onClick={[Function]}
    type="button"
    value="8"
  />
  <input
    className="button  "
    onClick={[Function]}
    type="button"
    value="7"
  />
  <input
    className="button  orange"
    onClick={[Function]}
    type="button"
    value="x"
  />
  <input
    className="button  "
    onClick={[Function]}
    type="button"
    value="6"
  />
  <input
    className="button  "
    onClick={[Function]}
    type="button"
    value="5"
  />
  <input
    className="button  "
    onClick={[Function]}
    type="button"
    value="4"
  />
  <input
    className="button  orange"
    onClick={[Function]}
    type="button"
    value="-"
  />
  <input
    className="button  "
    onClick={[Function]}
    type="button"
    value="3"
  />
  <input
    className="button  "
    onClick={[Function]}
    type="button"
    value="2"
  />
  <input
    className="button  "
    onClick={[Function]}
    type="button"
    value="1"
  />
  <input
    className="button  orange"
    onClick={[Function]}
    type="button"
    value="+"
  />
  <input
    className="button double "
    onClick={[Function]}
    type="button"
    value="0"
  />
  <input
    className="button  "
    onClick={[Function]}
    type="button"
    value="."
  />
  <input
    className="button  orange"
    onClick={[Function]}
    type="button"
    value="="
  />
</div>
`;
